### PR TITLE
Remove unnecessary DDC workaround

### PR DESCRIPTION
**Pull request by aaronlademann-wf**
## Motivation
#252 added a workaround for https://github.com/dart-lang/sdk/issues/36052 - which was fixed in Dart 2.2.1. 

> Because the 3.1.0 release will require at least Dart 2.4.0 for consumers, it is safe to remove the workaround.

## Changes
* Revert the builder changes that add the workaround
* Leave the regression tests intact to guard against further regressions in the Dart SDK

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: @greglittlefield-wf @corwinsheahan-wf @evanweible-wf @kealjones-wk @joebingham-wk 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        - Passing CI
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/component_declaration/component_base.dart [hunk 1]
```diff
@@ -518,27 +517,27 @@ abstract class UiProps extends MapBase
...
     List childArguments;
     // Use `identical` since it compiles down to `===` in dart2js instead of calling equality helper functions,
-    // and we don't want to allow any object overriding `operator==` to claim it's equal to `_notSpecified`.
-    if (identical(c1, notSpecified)) {
+    // and we don't want to allow any object overriding `operator==` to claim it's equal to `__notSpecified`.
```
Comment (evanweible-wf) [inline lib/src/component_declaration/component_base.dart]: [nit] find/replace changed this one to two underscores

File: lib/src/util/react_util.dart [hunk 2]
```diff
@@ -68,5 +67,6 @@ class UiPropsMapView extends MapView
...
+  // ignore: invalid_override_different_default_values_positional
+  ReactElement call([c1, c2, c3, c4, c5, c6, c7, c8, c9, c10, c11, c12, c13, c14, c15, c16, c17, c18, c19, c20, c21, c22, c23, c24, c25, c26, c27, c28, c29, c30, c31, c32, c33, c34, c35, c36, c37, c38, c39, c40]) => throw new UnimplementedError('@PropsMixin instances do not implement call');
```
Comment (evanweible-wf) [inline lib/src/util/react_util.dart]: Why are the `_notSpecified` default values still needed for `UiProps` but not this class?

Comment (aaronlademann-wf) [reply lib/src/util/react_util.dart]: I think they are actually @evanweible-wf - thus the failing unit tests.

Comment (aaronlademann-wf) [reply lib/src/util/react_util.dart]: I'm going to push a fix right now

**Comment by aaronlademann-wf** [comment]
Doesn't look like this is actually safe to revert.  Closing.

**Pull request closed without merging**